test/performance/range/container_seq_write_benchmark.cpp

```diff
@@ -95,13 +95,64 @@ BENCHMARK_TEMPLATE(sequential_write, seqan3::bitcompressed_vector, seqan3::aa27)
 BENCHMARK_TEMPLATE(sequential_write, seqan3::bitcompressed_vector, seqan3::alphabet_variant<char, seqan3::dna4>);
 
 BENCHMARK_TEMPLATE(sequential_write, small_vec, char);
+BENCHMARK_TEMPLATE(sequential_write, small_vec, uint32_t);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::gap);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::dna4);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::gapped<seqan3::dna4>);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::dna15);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::aa27);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::alphabet_variant<char, seqan3::dna4>);
 
+// ============================================================================
+//  SeqAn2: sequential_write
+// ============================================================================
+
+#if SEQAN3_HAS_SEQAN2
+
+#include <seqan/sequence.h>
+
+template <template <typename...> typename container_t, typename spec_t, typename alphabet_t>
+void sequential_write2(benchmark::State & state)
+{
+    auto cont_rando = seqan3::test::generate_sequence_seqan2<alphabet_t>(10'000, 0, 0);
+    container_t<alphabet_t, spec_t> source = cont_rando;
+
+    alphabet_t a{};
+    for (auto _ : state)
+    {
+        for (auto && c : source)
```

Comment:
```suggestion
        for (auto && range_value : source)
```

Reply:
~~I had a much longer answer that I deleted again.~~

Short and polite: "range_value" adds nothing in readability over "c". The line of code clearly states that "c" is a range element. It's like writing 
```c++
// This is a range-based-for-loop that loops over all elements of the range
```
before the line. Or writing:
```c++
for (int integer_counter = 0; integer_counter < 10; ++integer_counter)
```
For me these constructs make it harder to parse the code, because long names take inherently longer to read, they imply a certain complexity (complicated names are given to complicated things) or a meaning that is more global (it could even be read as a function parameter or a name from a namespace).

I agree that all names in the API should be well-reasoned and explanatory, but I think regulating naming down to this degree is -- politely -- a waste of everyone's time.

If there is strong consensus in the team that this is necessary, then I will change it and not spend more time discussing it, but since everyone always complains about the huge amounts of time that reviews take: maybe this is a good place to reduce "regulation".

Reply:
Not agreeing here, I think the basic question in naming anything is always "What is the thing I have here." and we should explicitly write it out. Not making any exceptions. I can flip your argument, if you named your stuff better, we wouldn't need to  decipher what you mean and could drastically decrease the review time.

In this case, with only reading that single line, I don't know what is happening here. `for (auto && c : source)` is telling me nothing.

We have two cases for naming:

1) If we don't know anything about the domain, as in fully generic code like `for_each(range, f)`. `range_value` is more expressive than a single letter. For example `c` might be short for a character and if I read it, I don't know whether you didn't know your domain (because it is fully generic) or if you knew it and used an abbreviation. We could give it a rule to allow a single letter which has no meaning, like for example `x`, but we rarely write functions like that.

2) We know the domain. For example the range is a (biological) sequence and hence the range value type is a character / alphabet. In this case we had two naming schemes `for (auto && chr : sequence)` or `for (auto && alphabet : sequence)` (I'm fine with both of them with slight preference for the first.) I encountered some examples in the library where we used `for (auto && c : s)` and these don't age well, because you don't immediately know what is going on.

Summing it up `range_value` would suggest me that you either knew the domain and were lazy to think of a proper name, or it was generic and you gave it a generic name.

My personal goal is that we should always try to reduce communication errors as much as possible and in this case we can  reduce 4 possible error classes

* domain unknown / non-descriptive name, (e.g. `for (auto && c : range)`)
* domain unknown / descriptive name, (e.g. `for (auto && range_value : range)`, `for (auto && value : range)`)
* domain known / non-descriptive name, (e.g. `for (auto && c : s)`)
* domain known / descriptive name (e.g. `for (auto && chr : sequence)`)

to 2 possible error classes

* domain unknown / descriptive name, (e.g. `for (auto && range_value : range)`, `for (auto && value : range)`)
* domain known / descriptive name (e.g. `for (auto && chr : sequence)`)

Just by being always descriptive.

Reply:
> My personal goal is that we should always try to reduce communication errors as much as possible 

Yes, I think we all want this. Just because someone disagrees on the "how" does not mean they are not interested in good communication or usability. But let's not drag this on for too long. I will change the names, but I seriously think this will make getting contributions more difficult as well as life harder for reviewers. Also, please add a guideline for this, so that contributors can look this up somewhere and it is not something that depends on the particular reviewer.

For the concrete names:
* `for (auto && range_value : range)` "range_value" is not typically a value but a reference. I will use `range_element` in the general case *if that is ok*.
* `for (auto && alphabet : sequence)` "alphabet" in SeqAn always refers to a type or gourp of types, and never to objects or references. -> there is no "sequence of alphabets" (unless you are iterating over type_list). If domain vocabulary is to be used (which I think is not bad in general), I will use `base : sequence`  *if that is ok*.

Reply:
> I will change the names, but I seriously think this will make getting contributions more difficult as well as life harder for reviewers. Also, please add a guideline for this, so that contributors can look this up somewhere and it is not something that depends on the particular reviewer.

Contributors are in most cases just okay with what someone suggests, and they don't argue against them.

> "range_value" is not typically a value but a reference. I will use range_element in the general case if that is ok.

True, but every reference should be convertible to a "range_value". `range_element` is a completely new word that we never used until now, but I like it. I'm arguing with myself if just `element` would be good enough.

> I will use `base : sequence` if that is ok

Again, we never used that term before. And a `base` is, at least for me, a base class. I thought this benchmark is for every alphabet instance, and we used "chr" throughout the library to name it.